### PR TITLE
CORE-1016: updates to support the event-recorder service.

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: amqp.go [hunk 1]
```diff
@@ -175,6 +180,31 @@ type TimeLimitRequest struct {
...
+}
+
+// NotificationMessage defines the structure of a notification message sent to
+// the Discovery Environment UI.
+type NotificationMessage struct {
+	Deleted       bool                   `json:"deleted"`
+	Email         bool                   `json:"email"`
+	EmailTemplate string                 `json:"email_template"`
+	Message       map[string]interface{} `json:"message"`
+	Payload       interface{}            `json:"payload"`
```
Comment (slr71) [inline amqp.go:201]: The legacy handler in the `event-recorder` service needs to be able to store a struct in the `Payload` field, so it uses interface. For the time being, the `Message` field always contains a map, so it's defined as a map in here for convenience. With the field defined as a map, we can examine its contents without having to do a type switch or a type assertion.

File: amqp.go [hunk 2]
```diff
@@ -601,13 +631,32 @@ func (c *Client) SetupPublishing(exchange string) error {
...
+// JSONPublishingOpts defines the set of publishing options used for JSON message
+// bodies.
+var JSONPublishingOpts = &PublishingOpts{
+	DeliveryMode: amqp.Persistent,
+	ContentType:  "application/json",
+}
+
+// PublishOpts sends a message to the configured exchange with options specified
+// in an options structure.
+func (c *Client) PublishOpts(key string, body []byte, opts *PublishingOpts) error {
```
Comment (slr71) [inline amqp.go:658]: The current notification agent uses a content type of `application/json` when publishing messages intended for consumption by the UI. This change allows callers to specify the content type or delivery mode without too much hassle.

File: amqp_test.go [hunk 3]
```diff
@@ -214,13 +211,13 @@ func TestSendStopRequest(t *testing.T) {
...
 	handler := func(d amqp.Delivery) {
-		d.Ack(false)
+		_ = d.Ack(false)
 		actual = d.Body
 		coord <- 1
 	}
 	key := StopRequestKey(invID)
 	client.AddConsumer(exchange(), exchangeType(), "test_queue4", key, handler, 0)
-	client.SendStopRequest(invID, "test_user", "this is a test")
+	_ = client.SendStopRequest(invID, "test_user", "this is a test")
```
Comment (ash3rz) [inline amqp_test.go:220]: I'm probably missing something, but what's the reason behind these small refactors?

**Review by ash3rz — APPROVED** [review]
Not an expert, but I didn't catch anything.  Just had a newbie question 👍

File: amqp_test.go (hunk 3 above)
Comment (slr71) [reply amqp_test.go:220]: Ah, that was just a small change to fix a linter warning. 🙂

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**